src/dialect/mod.rs

```diff
@@ -107,6 +253,10 @@ pub trait Dialect: Debug + Any {
         self.type_id()
     }
 
+    fn flags(&self) -> DialectFlags {
+        DialectFlags::default()
+    }
```

Comment:
I don't like creating a new struct on every call (and there are a lot).

I'd rather store the struct on each dialect, then the signature here can before

```rs
fn flags(&self) -> &DialectFlags {
  &self.settings
}
```

The problem then is that every struct needs a `fn new()` or impl `Default`, and the signature to create all dialects changes everywhere, but if we're happy with that breaking change, the whole thing would become much cleaner. WDYT?

Reply:
done.